class.jetpack-options.php

```diff
@@ -109,22 +127,27 @@ public static function update_option( $name, $value ) {
 	public static function update_options( $array ) {
 		$names = array_keys( $array );
 
-		foreach ( array_diff( $names, self::get_option_names(), self::get_option_names( 'non_compact' ) ) as $unknown_name ) {
+		foreach ( array_diff( $names, self::get_option_names(), self::get_option_names( 'non_compact' ), self::get_option_names( 'private' ) ) as $unknown_name ) {
 			trigger_error( sprintf( 'Invalid Jetpack option name: %s', $unknown_name ), E_USER_WARNING );
 			unset( $array[$unknown_name] );
 		}
 
-		foreach ( array_intersect( $names, self::get_option_names( 'non_compact' ) ) as $name ) {
-			update_option( "jetpack_$name", $array[$name] );
-			unset( $array[$name] );
+		foreach ( $names as $name ) {
+			self::update_option( $name, $array[$name] );
```

Comment:
Also, +lots to delegating the finding of the option to the subfunction, rather than sorting it out here.  Well done.